app-frontend/src/app/components/pages/project/layer/scenes/browse/index.js

```diff
@@ -1,11 +1,405 @@
+            let coords = _.map(coordsStrings, str => parseFloat(str));
+            // Leaflet expects nested coordinate arrays
+            bbox = [
+                [coords[1], coords[0]],
+                [coords[3], coords[2]]
+            ];
+        }
+        return bbox;
+    }
+
+    onRepositoryChange(bboxFetchFactory, repository) {
+        this.bboxFetchFactory = bboxFetchFactory;
+        this.currentRepository = repository;
+
+        this.sceneList = [];
+        this.sceneActions = new Map();
+        this.getMap().then((mapWrapper) => {
+            if (this.bboxCoords || this.queryParams && this.queryParams.bbox) {
+                this.fetchNextScenesForBbox = this.bboxFetchFactory(
+                    this.bboxCoords || this.queryParams.bbox
+                );
+                this.fetchNextScenes();
+            } else {
+                this.onViewChange(
+                    mapWrapper.map.getBounds(),
+                    mapWrapper.map.getCenter(),
+                    mapWrapper.map.getZoom()
+                );
+            }
+        });
+    }
+
+    onViewChange(newBounds, newCenter, zoom) {
+        // This gives 400 status code under planet data filter
+        // when the map extent is at a too big area
+        this.bboxCoords = newBounds.toBBoxString();
+        this.$parent.zoom = zoom;
+        this.$parent.center = newCenter;
+
+        this.setBboxParam(this.bboxCoords);
+
+        this.onBboxChange();
+    }
+
+    onViewChange(newBounds, newCenter, zoom) {
+        // This gives 400 status code under planet data filter
+        // when the map extent is at a too big area
+        this.bboxCoords = newBounds.toBBoxString();
+        this.zoom = zoom;
+        this.center = newCenter;
+
+        this.setBboxParam(this.bboxCoords);
+
+        this.onBboxChange();
+    }
+
+    onBboxChange() {
+        let queryParams = this.$location.search();
+        if (this.bboxFetchFactory && !queryParams.shape) {
+            this.fetchNextScenesForBbox = this.bboxFetchFactory(this.bboxCoords);
+            this.sceneList = [];
+            this.sceneActions = new Map();
+            this.debouncedSceneFetch();
+        }
+    }
+
+    isSelected(scene) {
+        return this.selected.has(scene.id);
+    }
+
+    onSelect(scene) {
+        if (this.selected.has(scene.id)) {
+            this.selected = this.selected.delete(scene.id);
+        } else {
+            this.selected = this.selected.set(scene.id, scene);
+        }
+        this.updateSelectText();
+    }
+
+    allVisibleSelected() {
+        let sceneSet = new Set(this.sceneList.map(s => s.id));
+        return this.selected.keySeq().toSet().intersect(sceneSet).size === sceneSet.size;
+    }
+
+    selectAll() {
+        if (this.allVisibleSelected()) {
+            this.selected = this.selected.clear();
+        } else {
+            this.selected = this.selected.merge(
+                _.filter(this.sceneList.map(s => [s.id, s]), s => !s.inLayer)
```

Comment:
I am not quite sure about what should be selected in here since when `this.selectText = 'Select all visible'` , should we select just the visible ones indicated by the eye icon, or should we select all in the `this.sceneList`? If the former, I think that we may be better map on `this.visibleScenes` instead in here.

Same for `allVisibleSelected()` function and related.

Reply:
visible here refers to stuff that's visible to the user in the list, not visible on the map. I'll try to come up with a better name

Reply:
ok. I would change the html text or add a tooltip @Lknechtli , up to you.